plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -1,61 +1,63 @@
 {-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE NamedFieldPuns       #-}
 {-# LANGUAGE OverloadedStrings    #-}
 {-# LANGUAGE UndecidableInstances #-}
 
 {- | Plutus conformance test suite library. -}
 module PlutusConformance.Common where
 
-import Control.DeepSeq (force)
+import Control.Lens (traverseOf)
 import Data.Text qualified as T
 import Data.Text.IO qualified as T
-import PlutusCore.Core (defaultVersion)
 import PlutusCore.Default (DefaultFun, DefaultUni)
-import PlutusCore.Error (ParserErrorBundle (ParseErrorB))
+import PlutusCore.Error (ParserErrorBundle)
 import PlutusCore.Evaluation.Machine.ExBudgetingDefaults (defaultCekParameters)
 import PlutusCore.Name (Name)
 import PlutusCore.Quote (runQuoteT)
 import PlutusPrelude
+import System.Directory
+import System.FilePath (takeBaseName, (</>))
 import Test.Tasty (defaultMain, testGroup)
-import Test.Tasty.Golden (findByExtension)
-import Test.Tasty.HUnit (testCase, (@=?))
 import Test.Tasty.Providers
 import Text.Megaparsec (SourcePos)
-import UnliftIO.Exception
+import UntypedPlutusCore qualified as UPLC
 import UntypedPlutusCore.Core.Type qualified as UPLC
 import UntypedPlutusCore.Evaluation.Machine.Cek (evaluateCekNoEmit)
 import UntypedPlutusCore.Parser qualified as UPLC
+import Witherable
 
--- Common functions for all tests
+type UplcProg = UPLC.Program Name DefaultUni DefaultFun ()
+
+type UplcEvaluator = UplcProg -> Maybe UplcProg
+
+-- | A UPLC evaluation test suite.
+data UplcEvaluationTest =
+    MkUplcEvaluationTest {
+       -- | The evaluator function to use for the test.
+       evaluator :: UplcEvaluator
+       -- | The test directory in which the test files are located.
+       , testDir :: FilePath
+    }
+
```

Comment:
Maybe add a bit of explanation here? If we like this we may want to add this to our style guide/policy about testing...I added a comment to plt-134.

Reply:
About the `IsTest` instance?